test/models/report_test.rb

```diff
@@ -3,7 +3,58 @@
 require 'test_helper'
 
 class ReportTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  setup do
+    @report = FactoryBot.create(:report)
+    @base_url = 'http://localhost:3000/reports'
+    @other_report_url = "#{@base_url}/#{FactoryBot.create(:report).id}"
```

Comment:
テストデータは無理にDRYにせず、あえてベタ書きする、という考え方もあります。
https://speakerdeck.com/jnchito/number-vstat

```suggestion
    @other_report_url = "http://localhost:3000/reports/#{FactoryBot.create(:report).id}"
```

Reply:
`@other_report`を使い回すのは許容範囲でしょうか。

Reply:
> `@other_report`を使い回すのは許容範囲でしょうか。

`@other_report_url`のことかな？
個人的にはこれなら許容範囲かなと思いました。
なぜなら、変数名がうまく情報を抽象化できているからです。
つまり、「この変数の中身は他のレポートのURLが入ってるんだな」と理解できます。

`"#{@base_url}/#{FactoryBot.create(:report).id}"`だと`@base_url`の中身が名前から想像できず、定義元を見に行きたくなる衝動に駆られました。こういうケースはNGかなと思います。

Reply:
> @other_report_urlのことかな？

記載間違い失礼しました。

> 変数名がうまく情報を抽象化できているからです。

脳内リソースを消費せずに中身が想像できるものについては変数化しても問題ないと理解しました！
テストを書いた後に上から下に読めるかの確認をしてリーダブルテストコードを意識していきたいと思います。